wa/workloads/antutu/__init__.py

```diff
@@ -109,16 +146,109 @@ def extract_scores(self, context, regex_version):
+                        if any([i in entry for i in ['Data', 'Document', 'Image', 'User', 'Video']]):
+                            ux.append(result)
+                            ux_result = sum(ux)
+                        if any([i in entry for i in ['RAM', 'ROM']]):
+                            mem.append(result)
+                            mem_result = sum(mem)
                         expected_results -= 1
         if expected_results > 0:
             msg = "The Antutu workload has failed. Expected {} scores, Detected {} scores."
             raise WorkloadError(msg.format(len(regex_version), expected_results))
 
+        context.add_metric('CPU Total Score', cpu_result, lower_is_better=False)
+        context.add_metric('GPU Total Score', gpu_result, lower_is_better=False)
+        context.add_metric('UX Total Score', ux_result, lower_is_better=False)
+        context.add_metric('MEM Total Score', mem_result, lower_is_better=False)
+
+        #Calculate overall scores
+        overall_result = float(cpu_result + gpu_result + ux_result + mem_result)
+        context.add_metric('Overall Score', overall_result, lower_is_better=False)
+
     def update_output(self, context):
         super(Antutu, self).update_output(context)
+        if self.version.startswith('10'):
+            self.extract_scores(context, self.regex_matches_v10)
         if self.version.startswith('9'):
             self.extract_scores(context, self.regex_matches_v9)
         if self.version.startswith('8'):
             self.extract_scores(context, self.regex_matches_v8)
         if self.version.startswith('7'):
             self.extract_scores(context, self.regex_matches_v7)
+
+
+class AntutuBDP(Workload):
+
+    name = "antutu_bdp"
+    description = '''
+    Workload for executing the BDP versions of the Antutu APK.
+
+    This will only work with specific APKS provided by Antutu but does
+    unlock command line automation and the capturing of a result file
+    as opposed to using UiAuto and Regex.
+
+    Known working version: 10.4.3-domesticAndroidFullBdp
+    '''
+    package_name = 'com.antutu.ABenchMark'
+    activity = 'com.android.module.app.ui.start.ABenchMarkStart --ez isExternal true --es whereTo "test"'
+
+    def initialize(self, context):
+        super(AntutuBDP, self).initialize(context)
+        #Set the files and directories we need
+        self.test_dir = os.path.join(self.target.external_storage_app_dir, 'com.antutu.ABenchMark/files/.antutu/')
+        self.settings_xml = context.get_resource(File(self, 'settings.xml'))
+        self.result_file = os.path.join(self.target.external_storage, 'Documents/antutu/last_result.json')
+        self.output_file = os.path.join(context.output_directory, 'antutu_results.json')
+
+    def setup(self, context):
+        super(AntutuBDP, self).setup(context)
+        #Install the supporting benchmark
+        supporting_apk = context.get_resource(ApkFile(self, package='com.antutu.benchmark.full'))
```

Comment:
Could we move the discovery of this APK to `__init__` as well? 
That would allow the workload to fail earlier if the required file can't be found.